capstone-rs/src/arch/x86.rs

```diff
@@ -218,8 +223,23 @@ impl Default for X86Operand {
 impl<'a> From<&'a cs_x86_op> for X86Operand {
     fn from(op: &cs_x86_op) -> X86Operand {
         let op_type = X86OperandType::new(op.type_, op.__bindgen_anon_1);
+
+        use cs_ac_type::*;
+        let access = if op.access == CS_AC_INVALID as _ {
+            RegAccessType::Invalid
+        } else if op.access == CS_AC_READ as _ {
+            RegAccessType::Read
+        } else if op.access == CS_AC_WRITE as _ {
+            RegAccessType::Write
+        } else if op.access == CS_AC_WRITE as u8 | CS_AC_READ as u8 {
+            RegAccessType::ReadWrite
+        } else {
+            unreachable!()
+        };
+
```

Comment:
Move this logic into an `impl From<cs_ac_type> for RegAccessType` in `instruction.rs`. That way, multiple architectures can use it.